### PR TITLE
update outdated  flag --version help text

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

To fix https://github.com/GoogleCloudPlatform/esp-v2/issues/679

### CODE REVIEW

File: docker/generic/start_proxy.py [hunk 1]
```diff
@@ -114,8 +114,7 @@ def make_argparser():
         '--version',
         default="",
         help=''' Set the service config ID of the Endpoints service.
-        If omitted and -c not specified, ESPv2 contacts the metadata
-        service to fetch the service config ID.  ''')
+        It is required if the "fixed" rollout strategy is used.''')
```
Comment (TAOXUY) [inline docker/generic/start_proxy.py:117]: Do we have unit test for this?

**Review by nareddyt — APPROVED** [review]

File: docker/generic/start_proxy.py (hunk 1 above)
Comment (qiwzhang) [reply docker/generic/start_proxy.py:117]: Done

File: docker/generic/start_proxy.py [hunk 2]
```diff
@@ -1011,12 +1010,14 @@ def make_argparser():
...
-          if args.version:
-            return "Flag --version cannot be used together with -R or --rollout_strategy."
-          if args.service_json_path:
+    if args.rollout_strategy == "managed":
+        if args.version:
+            return "Flag --version cannot be used if --rollout_strategy=managed."
+        if args.service_json_path:
             return "Flag -R or --rollout_strategy must be fixed with --service_json_path."
+    else:
+        if not args.version and not args.service_json_path:
```
Comment (TAOXUY) [inline docker/generic/start_proxy.py:1019]: no need check `not args.service_json_path`?

Comment (qiwzhang) [reply docker/generic/start_proxy.py:1019]: Yes, we need to check.  If service_json_path is specified,  version is NOT required.

**Review by TAOXUY — APPROVED** [review]

**Pull request merged**